File: 424. Longest Repeating Character Replacement/sol.go
package characterReplacement

func characterReplacement(s string, k int) int {
	chars := make(map[byte]int)
	res, maxf := 0, 0
	l := 0
	for r := range s {
		chars[s[r]]++
		if chars[s[r]] > maxf {
			maxf = chars[s[r]]
		}

		for (r-l+1)-maxf > k {
			chars[s[l]]--
			l++
		}
		if r-l+1 > res {
			res = r - l + 1
		}
	}

	return res

}

File: 424. Longest Repeating Character Replacement/sol.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        chars = {}
        res = 0
        l = 0
        maxf = 0
        for r in range(len(s)):
            chars[s[r]] = chars.get(s[r], 0) + 1
            maxf = max(maxf, chars[s[r]])
            # if (r - l + 1) - max(chars.values()) > k:
            if (r - l + 1) - maxf > k:
                chars[s[l]] -= 1
                l += 1
            res = max(r - l + 1, res)
        return res



